### PR TITLE
2552 Fix ToNumpy issue with 0-dim data

### LINKED ISSUE #2552: ToNumpy changed the dim of 0-dim data
**Describe the bug**
Holger found an issue that the `ToNumpy` transform will add unexpected dim to 0-dim data in classification task.
I think it's similar issue as: https://github.com/Project-MONAI/MONAI/pull/2454.

**How to reproduce**
```
>>> import numpy as np
>>> from monai.transforms import ToNumpy, ToTensor
>>> 
>>> print(ToNumpy()(5))
[5]
```

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2552 .

### Description
This PR enhanced the `ToNumpy` transform to support 0-dim data.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_to_numpy.py [hunk 1]
```diff
@@ -58,6 +58,13 @@ def test_list_tuple(self):
         result = ToNumpy()(test_data)
         np.testing.assert_allclose(result, np.asarray(test_data))
 
+    def test_single_value(self):
+        for test_data in [5, np.array(5)]:
```
Comment (yiheng-wang-nv) [inline tests/test_to_numpy.py:62]: suggest to add `torch.tensor(5)`

**Review by yiheng-wang-nv — APPROVED** [review]

File: tests/test_to_numpy.py (hunk 1 above)
Comment (Nic-Ma) [reply tests/test_to_numpy.py]: Sure, updated in the latest commit.

Thanks.
